src/Components/Search/NewSearch/Mobile/Overlay.tsx

```diff
@@ -87,37 +83,57 @@ export const Overlay: FC<OverlayProps> = ({ viewer, relay, onClose }) => {
 
   const handlePillClick = (pill: PillType) => {
     setSelectedPill(pill)
-    refetch(value, pill.searchEntityName)
+    refetch(inputValue, pill.searchEntityName)
+
+    scrollToTop()
   }
 
   const handleValueChange = event => {
-    const inputValue = event.target.value
-    setValue(inputValue)
+    const value = event.target.value
+    setInputValue(value)
 
-    if (shouldStartSearching(inputValue))
-      debouncedSearchRequest(inputValue, selectedPill.searchEntityName)
+    if (shouldStartSearching(value))
+      debouncedSearchRequest(value, selectedPill.searchEntityName)
   }
 
   return (
-    <OverlayBase onClose={onClose}>
-      <Box mt={-15}>
-        <LabeledInput
-          ref={inputRef}
-          value={value}
-          placeholder={t`navbar.searchArtsy`}
-          label={<SearchIcon fill="black60" aria-hidden mr={-10} size={18} />}
-          onChange={handleValueChange}
-          // px={2}
+    <OverlayBase
+      onClose={onClose}
+      header={
+        <>
+          <Box mt={-15}>
+            <LabeledInput
+              mx={2}
+              ref={inputRef}
+              value={inputValue}
+              placeholder={t`navbar.searchArtsy`}
+              label={
+                <SearchIcon fill="black60" aria-hidden mr={-10} size={18} />
```

Comment:
**suggestion**: do we really want (and need) to use `mr={-10}` here?

Reply:
I know I say this _a lot_ but really: please don't fight against what Palette proscribes. If the spec is intentional then frequently it is an oversight. It may be the case that we need a labelled input that has an icon that is flush with the edge but if that was the case then if the component is updated this instance will break.